### PR TITLE
Fix ending number as time

### PR DESCRIPTION
**Pull request by iamnwi**
fix case "move 3pm appointment to 4"

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishTimeExtractorConfiguration.cs [hunk 1]
```diff
@@ -97,6 +98,10 @@ public class EnglishTimeExtractorConfiguration : ITimeExtractorConfiguration
                 DateTimeDefinitions.ConnectNumRegex,
                 RegexOptions.IgnoreCase | RegexOptions.Singleline);
 
+        public static readonly Regex SpecialTimePattern =
+            new Regex(DateTimeDefinitions.SpecialTimePattern,
+                RegexOptions.IgnoreCase | RegexOptions.Singleline);
```
Comment (rubio41) [inline .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishTimeExtractorConfiguration.cs]: Rename it to NumberEndingPattern, the same to all SpecialTimePattern.
This is not a good naming.

File: .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishTimeExtractorConfiguration.cs [hunk 2]
```diff
@@ -140,10 +145,16 @@ public class EnglishTimeExtractorConfiguration : ITimeExtractorConfiguration
 
         IEnumerable<Regex> ITimeExtractorConfiguration.TimeRegexList => TimeRegexList;
 
+        public IExtractor TimeExtractor => new BaseTimeExtractor(new English.EnglishTimeExtractorConfiguration());
+
+        public IExtractor NumExtractor => NumberExtractor.GetInstance();
```
Comment (rubio41) [inline .NET/Microsoft.Recognizers.Text.DateTime/English/Extractors/EnglishTimeExtractorConfiguration.cs]: These should be in the construction method. Please follow the initialization of the durationExtractor in this file.

File: .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseTimeExtractor.cs [hunk 3]
```diff
@@ -68,15 +70,34 @@ private List<Token> AtRegexMatch(string text)
...
+                matches = this.config.IshRegex.Matches(text);
+
                 foreach (Match match in matches)
                 {
                     ret.Add(new Token(match.Index, match.Index + match.Length));
                 }
             }
+            // handle special patterns
+            //Regex r = new Regex($@"\b((?<oldTime>.*)\s+(?<meeting>meeting|appointment|conference|assembly)\s+to\s+(?<newTime>.*)(.|,)?)\b",
+            //RegexOptions.IgnoreCase | RegexOptions.Singleline);
```
Comment (rubio41) [inline .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseTimeExtractor.cs]: Please delete the commented code.

File: .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseTimeExtractor.cs [hunk 4]
```diff
@@ -68,15 +70,34 @@ private List<Token> AtRegexMatch(string text)
...
+            matches = this.config.SpecialTimePattern.Matches(text);
+            foreach (Match match in matches)
+            {
+                // check if oldTime string is a time format
+                var oldTime = match.Groups["oldTime"];
+                var timeRes = this.config.TimeExtractor.Extract(oldTime.ToString());
+                if (timeRes.Count == 0) continue;
```
Comment (rubio41) [inline .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseTimeExtractor.cs]: if (timeRes.Count == 0) 
{
continue;
}

always using {} after "if"

File: .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseTimeExtractor.cs [hunk 5]
```diff
@@ -68,15 +70,34 @@ private List<Token> AtRegexMatch(string text)
...
+                // check if newTime string is a number format
+                var newTime = match.Groups["newTime"];
+                var numRes = this.config.NumExtractor.Extract(newTime.ToString());
+                if (numRes.Count == 0) continue;
```
Comment (rubio41) [inline .NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseTimeExtractor.cs]: same as above

File: .NET/Microsoft.Recognizers.Text.DateTime/Extractors/IDateTimeExtractor.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+﻿// Enable a reference time pass to the extractor
```
Comment (rubio41) [inline .NET/Microsoft.Recognizers.Text.DateTime/Extractors/IDateTimeExtractor.cs]: This should be deleted since we do the IDatetimeExtractor in another PR.

File: .NET/Microsoft.Recognizers.Text.DateTime/French/Extractors/FrenchTimeExtractorConfiguration.cs [hunk 7]
```diff
@@ -172,9 +177,16 @@ public class FrenchTimeExtractorConfiguration : ITimeExtractorConfiguration
             ConnectNumRegex
         };
         Regex ITimeExtractorConfiguration.IshRegex => IshRegex;
+
+        public IExtractor TimeExtractor => new BaseTimeExtractor(new English.EnglishTimeExtractorConfiguration());
+
+        public IExtractor NumExtractor => new NumberExtractor();
+
```
Comment (rubio41) [inline .NET/Microsoft.Recognizers.Text.DateTime/French/Extractors/FrenchTimeExtractorConfiguration.cs]: same as above

File: .NET/Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishTimeExtractorConfiguration.cs [hunk 8]
```diff
@@ -89,10 +92,16 @@ public class SpanishTimeExtractorConfiguration : ITimeExtractorConfiguration
 
         Regex ITimeExtractorConfiguration.IshRegex => null;
 
+        public IExtractor TimeExtractor => new BaseTimeExtractor(new English.EnglishTimeExtractorConfiguration());
```
Comment (rubio41) [inline .NET/Microsoft.Recognizers.Text.DateTime/Spanish/Extractors/SpanishTimeExtractorConfiguration.cs]: same as above

**Review by rubio41 — CHANGES_REQUESTED** [review]

File: .NET/Microsoft.Recognizers.Text.DateTime/Extractors/IDateTimeExtractor.cs (hunk 6 above)
Comment (iamnwi) [reply .NET/Microsoft.Recognizers.Text.DateTime/Extractors/IDateTimeExtractor.cs]: I didn't see it in my project, maybe it still exist in folder, do I need to delete it in folder also?
